lib/ruby_lsp/tapioca/addon.rb

```diff
@@ -60,6 +60,22 @@ def version
 
       sig { params(changes: T::Array[{ uri: String, type: Integer }]).void }
       def workspace_did_change_watched_files(changes)
+        constants = changes.filter_map do |change|
+          path = change[:uri].gsub("file://", "")
+
+          entries = T.must(@index).entries_for(path, RubyIndexer::Entry::Class, RubyIndexer::Entry::Module)
+          next unless entries
+        end.flatten
+
+        return if constants.empty?
+
+        T.must(@rails_runner_client).send_notification("reload", {})
+        T.must(@rails_runner_client).send_notification(
+          "server_addon/delegate",
+          request_name: "dsl",
+          server_addon_name: "Tapioca",
+          constants: constants,
+        )
```

Comment:
We should add a proper API for this in the Rails add-on. Something like `@rails_runner_client.delegate()`.